File: web/src/app/core/models/question-box-entry.model.ts
export class QuestionBoxEntryModel {
    BoxId?: string;
    EntryId? : string;
    Code?: string;
    Question?: string;
    Upvotes?: number; 
    Downvotes?: number;
    IsHidden?: boolean;
    UserUpvoted?: boolean;
    UserDownvoted?: boolean;
    CreatedAt? : Date;
    CreatedBy?: number;
} 

File: GO/core/model/questionboxentry/questionboxentry.model.go
package questionboxentrymodel

import (
	"time"

	uuid "github.com/satori/go.uuid"
)

// QuestionBoxEntry is a response to a question box
type QuestionBoxEntry struct {
	EntryId    uuid.UUID
	BoxId      uuid.UUID
	Question   string
	IsHidden   bool
	IsFavorite bool
	Upvotes    int
	Downvotes  int
	CreatedAt  time.Time
	CreatedBy  string
	UpdatedAt  *time.Time
	UpdatedBy  string
	DeletedAt  *time.Time
}

// TableName returns the table name for use with ORM
func (QuestionBoxEntry) TableName() string {
	return "question_box_entries"
}
